server/src/api/v1.py

```diff
@@ -156,27 +194,25 @@ def job_builder(data: dict, auth_token: str):
     if has_attachments(data):
         data["attachments_status"] = "waiting"
 
-    if "job_priority" in data:
-        priority_level = data["job_priority"]
-        job_queue = data["job_queue"]
-        allowed = check_token_priority_permission(
-            auth_token,
-            os.environ.get("JWT_SIGNING_KEY"),
-            priority_level,
-            job_queue,
+    priority_level = data.get("job_priority", 0)
+    job_queue = data["job_queue"]
+    allowed = check_token_permissions(
+        auth_token,
+        os.environ.get("JWT_SIGNING_KEY"),
+        priority_level,
+        job_queue,
+    )
+    if not allowed:
+        abort(
+            403,
+            (
+                f"Not enough permissions to push to {job_queue}",
+                f"with priority {priority_level}",
+            ),
         )
-        if not allowed:
-            abort(
-                403,
-                (
-                    f"Not enough permissions to push to {job_queue}",
-                    f"with priority {priority_level}",
-                ),
-            )
-        job["job_priority"] = priority_level
-        data.pop("job_priority")
-    else:
-        job["job_priority"] = 0
+    job["job_priority"] = priority_level
+    data.pop("job_priority", None)
```

Comment:
Why do we pop here?

Reply:
I don't think it's strictly necessary - I pop only so that job_priority isn't duplicated in both the job dict and the data dict. Is there a reason to include it in the data dictionary as well?

Reply:
but the data dict is just the data that's passed in from the original json job submission right? so afaict it's not used anywhere after that, so modifying it is just removing a key from data that you're about to throw away... unless I'm not seeing something?

Reply:
The data gets added to the job dict right? In line 217